qiime/core/result_base.py

```diff
@@ -26,6 +28,14 @@ def load(cls, filepath):
         result._assert_valid_type(result.type)
         return result
 
+    @classmethod
+    def extract(cls, filepath, output_dir=None):
```

Comment:
👍

I'd prefer this method to exist on `Archiver` since it deals with zipfiles and `Archiver` will be responsible for handling multiple archive formats when they exist (and they may not always be zip-based). Can you move it there and create an `extract` method here that defers?